openapi-v1.yaml

```diff
@@ -6156,3 +6278,124 @@ paths:
           description: error response
           schema:
             $ref: "#/definitions/Error"
+
+  /taskgraphs/log:
```

Comment:
Almost all our endpoints use namespace in the url like `/taskgraphs/{namespace}/log` and we use the code in middleware for access control. If you pass namespace in the body it won't be uniform and you will have to copy paste for auth. Can you comment on the decision?

Reply:
I chose this by analogy to the way `ArrayTask` logs work, where array tasks do not have a particular user associated with them at the URL level. This makes sense to me because I see the case where a user would want to look at all the task graph logs they have access to (much like they look at all the array tasks they have access to). I am not hugely attached to this decision. @Shelnutt2 any thoughts on the matter?

Reply:
Tasks are not posted right? We usually post to namespaces as @anastasop highlights so that we can check with the model enforcer middleware for permissions.

Also `/log` does not makes sense here. This isn't posting a log, its posting the graph to run no?

Reply:
The entity that is being created is just the log of the run (or the skeleton of the log), not the run itself.

The other difficulty of including the namespace directly in the URL is that when a task is submitted, we still have to verify that the user that submitted the task is able to make updates to the task graph anyway, because separate jobs in a task graph can be run from individual namespaces. This means that including the namespace in the URL would not actually save any complexity:

**Setup**

 0. User A is a member of namespaces N and P. User B is a member of only P.
 1. User A creates task graph G in namespace N

**Scenarios**

- User A executes a task from G under namespace N.
  - we want to update graph G, and we already know that A has permission for N.
- User A executes a task from G under namespace P
  - we still want to update graph G, and we need to check that A has permission for N.
- User A executes a task from G under their own namespace
  - we still want to update graph G, and we need to check that A has permission for N.
- User B executes a task and spoofs it as belonging to G
  - we do not want to update graph G, and we need to check that B has (lacks) permission for N.

We could eliminate this complexity if we decide that “only tasks run under the same namespace as the graph is registered to will be included in the log”, though.

Reply:
After discussion, @antalakas and I have decided that only tasks run under *the same namespace* as the graph will be included in a log. This will be updated and reflected in a forthcoming version of this change.

Reply:
@thetorpedodog @Shelnutt2 @antalakas Changed the endpoints to use `{namespace}` params. Can you give another review?

Reply:
Having the namespace in the path for post is an established pattern since in this way we allow model enforcer to do the checks. This also aligns with the decision to only deal with tasks under the same namespace as the graph

I think we could unify endpoint `/taskgraphs/{namespace}` for post and fetch all without using `/logs` or `/log` in the end
Same for patch , get, i think we can eliminate `/logs` and just keep `/id`

Reply:
`/logs` is a deliberate choice to allow for namespacing task graph logs versus submission of server-side task graphs.

Reply:
The discussion about the use of namespaces is finalized at https://app.shortcut.com/tiledb-inc/story/13280/implement-endpoints-for-task-graph-log-reporting-api